sdk/identity/azure-identity/azure/identity/__init__.py

```diff
@@ -53,6 +54,7 @@ def __init__(self, **kwargs):
 
 
 __all__ = [
+    "AuthorizationCodeCredential",
```

Comment:
If `AuthorizationCodeCredential` is going in the public API for `azure.identity` why does it need to be in a "private" module `_credentials`?

Reply:
To control documentation. Users should write
```py
from azure.identity import AuthorizationCodeCredential
```
but our published documentation includes all "public" modules. So if `AuthorizationCodeCredential` is defined in a "public" module, it's documented twice, once as part of `azure.identity` (where it's imported and included in `__all__`) and again as part of `azure.identity.credentials`. This is the status quo on [azure.github.io](https://azure.github.io/azure-sdk-for-python/ref/azure.identity.html).

Defining classes in a "private" module is an easy way to prevent that. I plan to move all the credentials into `_credentials`.